enhancements/subscription-content/subscription-content-access.md

```diff
@@ -0,0 +1,242 @@
+containers
+
+### Implementation Details/Notes/Constraints [optional]
+
+To consume RHEL content in OpenShift, the running container must have access to the following:
+
+One or more entitlement keys, which are PEM-encoded certificates. A yum repository definition, which
+tells yum/dnf where to access content from.
+
+Yum repository definitions are specific to the major version of RHEL that the running container is
+based on - a RHEL 8-based container is not compatible with RHEL 7 content.
+
+This enhancement proposal will detail at a high level the functions of:
+
+A cluster-wide subscription manager to generate subscription content access data A projected
+resource CSI driver to share resources across OpenShift namespaces A subscription content webhook to
+generate the necessary volume mounts to add subscription content access data to containers
+Extensions to propagate subscription content access data into OpenShift builds.
+
+#### Cluster Subscription Manager
+
+The cluster subscription manager will take as input the `cloud.redhat.com` pull secret, which
+already provides authentication for the cluster and provides the means to register clusters with Red
+Hat’s cloud manager. The pull secret is then used to obtain the following: The `redhat.repo` yum
+repo configurations for all supported major versions of RHEL. Today these will be RHEL 7 and 8 -
+later major versions of RHEL must be anticipated. The Simple Content Access certificate for the
+customer, which will serve as the entitlement key.
+
+As RHSM does today on RHEL hosts, the cluster subscription manager will be responsible for
+generating `redhat.repo` definitions for the currently supported RHEL major versions and saving it
+on the cluster. Each `redhat.repo` file will be saved in a ConfigMap within the `openshift-config`
+namespace, named `redhat-repo-rhel${n}`, where `${n}` is the major version of RHEL associated with
+the `redhat.repo` file.
+
+The cluster subscription manager will also need to obtain the entitlement keys associated with the
+attached OpenShift subscription. It is assumed that OpenShift entitlements grant access to RHEL
+content, but may not grant access to layered content such as EAP or Application Runtimes. This
+content must be saved in a Secret within the `openshift-config` namespace, named
+`etc-pki-entitlement`.
+
+The full details of this component will be specified in a subsequent enhancement proposal.
+
+#### Projected-resource CSI Driver
+
+Once the entitlement key(s) and yum repository definitions are stored on the cluster, the next step
+is to share these definitions across namespaces. This task will fall to the projected-resource
+containers storage interface (CSI) driver. This will be a special storage driver that allows
+ConfigMaps and Secrets to be shared across namespaces via standard pod volume mounts. Access to
+shared Secrets and ConfigMaps will be controlled via namespace-scoped Custom Resources. The
```

Comment:
Actually I think it is more likely the `ProjectedResource` CR would be cluster scoped

I would change this to 

`... controlled via RBAC applied to various instances of the new CustomResource type ProjectedResource`

Reply:
Working through the webhook piece, things work better if the CR is cluster-scoped. Will update.